src/main/java/com/insightfullogic/java8/answers/chapter5/Fibonacci.java

```diff
@@ -8,7 +8,7 @@
     private final Map<Integer,Long> cache;
 
     public Fibonacci() {
-        cache = new HashMap<>();
+        cache = new ConcurrentHashMap<>();
```

Comment:
`ConcurrentHashMap` is useless in this context as the fibonacci sequence is evaluated inside a single thread.

Remember, recursive evaluation has no relationship with parallel computing.

Reply:
I does help, I just checked it with Java 14. HashMap doesn't work while ConcurrentHashMap does. See my comment below.